index.bs

```diff
@@ -1032,29 +1079,43 @@ To
 given a [=file entry=] |file| in a [=/Realm=] |realm|:
 
 1. Let |stream| be a [=new=] {{FileSystemWritableFileStream}} in |realm|.
-1. Set |stream|.[=FileSystemWritableFileStream/[[file]]=] to |file|.
+1. Set |stream|'s [=FileSystemWritableFileStream/[[file]]=] to |file|.
 1. Let |writeAlgorithm| be an algorithm which takes a |chunk| argument
    and returns the result of running the [=write a chunk=] algorithm with |stream| and |chunk|.
 1. Let |closeAlgorithm| be these steps:
    1. Let |closeResult| be [=a new promise=].
-   1. Run these steps [=in parallel=]:
-      1. Let |access| be the result of running |file|'s [=file system entry/query access=] given "`readwrite`".
-      1. If |access| is not "{{PermissionState/granted}}",
-         [=/reject=] |closeResult| with a "{{NotAllowedError}}" {{DOMException}}
-         and abort.
-      1. Run [=implementation-defined=] malware scans and safe browsing checks.
-         If these checks fail, [=/reject=] |closeResult| with an "{{AbortError}}" {{DOMException}} and abort.
-      1. Set |stream|.[=FileSystemWritableFileStream/[[file]]=]'s [=file entry/binary data=] to |stream|.[=[[buffer]]=].
-         If that throws an exception, [=/reject=] |closeResult| with that exception and abort.
-
-         Note: It is expected that this atomically updates the contents of the file on disk
-         being written to.
-
-      1. [=file entry/lock/release|Release the lock=] on |stream|.[=FileSystemWritableFileStream/[[file]]=].
-      1. [=/Resolve=] |closeResult| with `undefined`.
+   1. [=Enqueue the following steps=] to the [=file system queue=]:
+      1. Let |access| be the result of running |file|'s
+         [=file system entry/query access=] given "`readwrite`".
+      1. If |access| is not "{{PermissionState/granted}}":
+         1. Set |requestAccessError| to |access| if |access| is an
+            [=exception/error name=]; otherwise, "{{NotAllowedError}}".
+         1. [=Queue a storage task=] with |file|'s [=relevant global object=] to
+            [=/reject=] |closeResult| with a
+            |requestAccessError| {{DOMException}} and abort these steps.
+
+      1. [=Queue a storage task=] with |file|'s [=relevant global object=]
+         to run these steps:
+        1. Run [=implementation-defined=] malware scans and safe browsing checks.
+           If these checks fail, [=/reject=] |closeResult| with an
+           "{{AbortError}}" {{DOMException}} and abort these steps.
+        1. Set |stream|'s [=FileSystemWritableFileStream/[[file]]=]'s
+           [=file entry/binary data=] to |stream|'s [=[[buffer]]=].
+           If that throws an exception, [=/reject=] |closeResult| with that
+           exception and abort these steps.
+
+           Note: It is expected that this atomically updates the contents of the
+           file on disk being written to.
+
+        1. [=file entry/lock/release|Release the lock=] on
+           |stream|'s [=FileSystemWritableFileStream/[[file]]=].
```

Comment:
Looking at this again - since we're running in a "storage task" at this point (albeit one posted from the "file system queue"), do we need to explicitly queue this to the "file system queue"?

Reply:
Yes. Because these steps run on the main thread, not the "file system queue" thread.